### Case count in Texas

In [2]:
%%bigquery
select * from covid_19_staging.cases_SQL_Final_Date where state = "Texas" and country = "US" order by date

state country        date  confirmed  deaths  recovered
0   Texas      US  2020-03-10         13       0          0
1   Texas      US  2020-03-11         21       0          0
2   Texas      US  2020-03-11         43       0          0
3   Texas      US  2020-03-12         27       0          0
4   Texas      US  2020-03-14         57       0          0
5   Texas      US  2020-03-15         72       0          0
6   Texas      US  2020-03-16         85       0          0
7   Texas      US  2020-03-17        110       1          0
8   Texas      US  2020-03-18        173       3          0
9   Texas      US  2020-03-19        260       5          0
10  Texas      US  2020-03-20        394       5          0
11  Texas      US  2020-03-21        581       5          0
12  Texas      US  2020-03-30        563       5          0
13  Texas      US  2020-03-31        680       6          0
14  Texas      US  2020-04-01        847       6          0
15  Texas      US  2020-04-02        955       6          0
16  Texas      US  2020-04-03       1106      13          0
17  Texas      US  2020-04-04       1284      17          0
18  Texas      US  2020-04-05       1395      20          0
19  Texas      US  2020-04-06       1809      22          0
20  Texas      US  2020-04-07       2146      23          0
21  Texas      US  2020-04-08       2341      31          0
22  Texas      US  2020-04-09       3047      34          0
23  Texas      US  2020-04-10       3261      40          0
24  Texas      US  2020-04-11       3561      41          0
25  Texas      US  2020-04-12       3629      44          0
26  Texas      US  2020-04-13       3744      46          0
27  Texas      US  2020-04-14       3907      52          0
28  Texas      US  2020-04-15       4097      58          0
29  Texas      US  2020-04-16       4306      63          0
30  Texas      US  2020-04-17       4460      67          0
31  Texas      US  2020-04-18       4653      71          0
32  Texas      US  2020-04-19       4823      75          0
33  Texas      US  2020-04-20       4977      78          0
34  Texas      US  2020-04-21       5143      79          0
35  Texas      US  2020-04-22       5211      80          0
36  Texas      US  2020-04-24       5330      82          0
37  Texas      US  2020-04-25       5482      84          0
38  Texas      US  2020-04-26       5482      84          0
39  Texas      US  2020-04-27       5628      84          0
40  Texas      US  2020-04-28       5628      84          0
41  Texas      US  2020-04-29       5827      98          0
42  Texas      US  2020-04-30       6161     109          0
43  Texas      US  2020-05-01       6356     114          0
44  Texas      US  2020-05-02       6551     122          0
45  Texas      US  2020-05-03       6708     129          0
46  Texas      US  2020-05-04       6838     133          0
47  Texas      US  2020-05-05       6967     140          0
48  Texas      US  2020-05-06       7128     144          0
49  Texas      US  2020-05-07       7244     150          0
50  Texas      US  2020-05-08       7377     154          0
51  Texas      US  2020-05-09       7594     160          0

### The daily increase of confirmed cases in Texas

In [34]:
%%bigquery
with cte as(select ROW_NUMBER() over(ORDER BY date) as date_rank, date, sum(confirmed) as confirmed,
            FROM covid_19_staging.cases_SQL_Final_Date
            where state = "Texas" and country = "US"
           group by date)
SELECT c1.*, c1.confirmed - c2.confirmed  as daily_growth
FROM cte c1
LEFT JOIN cte c2 ON c2.date_rank = c1.date_rank - 1
order by daily_growth desc limit 5

date_rank        date  confirmed  daily_growth
0         22  2020-04-09       3047           706
1         19  2020-04-06       1809           414
2         20  2020-04-07       2146           337
3         42  2020-04-30       6161           334
4         24  2020-04-11       3561           300

In [36]:
%%bigquery
select daily_growth_rate from (
with cte as(select ROW_NUMBER() over(ORDER BY date) as date_rank, date, sum(confirmed) as confirmed,
            FROM covid_19_staging.cases_SQL_Final_Date
            where state = "Texas" and country = "US"
           group by date)
SELECT c1.*, round((c1.confirmed - c2.confirmed)*100/c2.confirmed,2)  as daily_growth_rate
FROM cte c1
LEFT JOIN cte c2 ON c2.date_rank = c1.date_rank - 1)
where date = "2020-04-09"

daily_growth_rate
0              30.16

Therefore, on April 9th Texas has the most daily growth, with a growth rate of 30.16%

visualization - case count trend in Texas

In [40]:
%%bigquery
create view reporting.v_case_count_trend_Texas as
select state,date,confirmed,deaths,recovered  from 
myproject60392.covid_19_staging.cases_SQL_Final_Date 
where state = "Texas" and country = "US" order by date

Empty DataFrame
Columns: []
Index: []

visualization - mobility trend in Texas

In [41]:
%%bigquery
create view reporting.v_mobility_trend_Texas as
select * from 
myproject60392.apple_mobility_staging.mobility_table
where region = "Texas" 

Empty DataFrame
Columns: []
Index: []

May 7 case count nationalwide

In [92]:
%%bigquery
create view reporting.v_case_count_map_US as
select state,date,confirmed,deaths,recovered  from 
myproject60392.covid_19_staging.cases_SQL_Final_Date 
where country = "US" and date ="2020-05-07" order by date

Empty DataFrame
Columns: []
Index: []

visualization - case count trend in the US

In [ ]:
%%bigquery
create view reporting.v_case_count_trend_US as
select state,date,confirmed,deaths,recovered  from 
myproject60392.covid_19_staging.cases_SQL_Final_Date 
where country = "US"  date = order by date

### case count in Hubei vs other part of china

In [3]:
%%bigquery
select * from covid_19_staging.cases_SQL_Final_Date
where country in ('Mainland China', 'China') 

state         country        date  confirmed  deaths  recovered
0              Anhui           China  2020-05-09        991       6        985
1              Anhui           China  2020-05-07        991       6        985
2              Anhui           China  2020-05-04        991       6        985
3              Anhui           China  2020-03-08        990       6        984
4              Anhui           China  2020-04-30        991       6        985
...              ...             ...         ...        ...     ...        ...
2413  Inner Mongolia  Mainland China  2020-02-07         50       0          5
2414  Inner Mongolia  Mainland China  2020-02-10         58       0          5
2415  Inner Mongolia  Mainland China  2020-02-05         42       0          3
2416  Inner Mongolia  Mainland China  2020-02-25         75       0         35
2417  Inner Mongolia  Mainland China  2020-01-26          7       0          0

[2418 rows x 6 columns]

In [4]:
%%bigquery
select "Hubei" as province, date,sum(confirmed) as confirmed, sum(deaths) as deaths, sum(recovered) as recovered
from covid_19_staging.cases_SQL_Final_Date
where state = "Hubei"
group by date
union all
select "Provinces other than Hubei" as province, date,sum(confirmed) as confirmed, sum(deaths) as deaths, sum(recovered) as recovered
from covid_19_staging.cases_SQL_Final_Date
where country in ('Mainland China', 'China')  and state != "Hubei"
group by date
order by date

province        date  confirmed  deaths  recovered
0    Provinces other than Hubei  2020-01-22        103       0          0
1                         Hubei  2020-01-22        444      17         28
2    Provinces other than Hubei  2020-01-23        195       1          2
3                         Hubei  2020-01-23        444      17         28
4    Provinces other than Hubei  2020-01-24        367       2          5
..                          ...         ...        ...     ...        ...
198                       Hubei  2020-05-07      68128    4512      63616
199  Provinces other than Hubei  2020-05-08      15847     125      15361
200                       Hubei  2020-05-08      68128    4512      63616
201  Provinces other than Hubei  2020-05-09      15848     125      15377
202                       Hubei  2020-05-09      68128    4512      63616

[203 rows x 5 columns]

### The daily increase of confirmed cases in China

In [6]:
%%bigquery
create table covid_19_staging.cases_in_China as
select t2.state, t1.date_rank, t2.date,t2.confirmed from(
select ROW_NUMBER() over(ORDER BY date) as date_rank, date
from covid_19_staging.cases_SQL_Final_Date
where country in ('Mainland China', 'China')
group by date) t1
right join 
(select state, date, confirmed from covid_19_staging.cases_SQL_Final_Date
where country in ('Mainland China', 'China')) t2
on t1.date = t2.date 
order by date

Empty DataFrame
Columns: []
Index: []

In [7]:
%%bigquery
select * from covid_19_staging.cases_in_China limit 6

state  date_rank        date  confirmed
0  Anhui          1  2020-01-22          1
1  Gansu          1  2020-01-22          0
2  Hebei          1  2020-01-22          1
3  Henan          1  2020-01-22          5
4  Hubei          1  2020-01-22        444
5  Hunan          1  2020-01-22          4

In [8]:
%%bigquery
WITH cte AS(SELECT state, date_rank, date, confirmed,
            FROM covid_19_staging.cases_in_China )
SELECT c1.*, c1.confirmed - c2.confirmed as daily_growth
FROM cte c1
LEFT JOIN cte c2 ON c1.state = c2.state AND c2.date_rank = c1.date_rank - 1
limit 50

state  date_rank        date  confirmed  daily_growth
0            Anhui          1  2020-01-22          1           NaN
1            Gansu          1  2020-01-22          0           NaN
2            Hebei          1  2020-01-22          1           NaN
3            Henan          1  2020-01-22          5           NaN
4            Hubei          1  2020-01-22        444           NaN
5            Hunan          1  2020-01-22          4           NaN
6            Jilin          1  2020-01-22          0           NaN
7            Tibet          1  2020-01-22          0           NaN
8           Fujian          1  2020-01-22          1           NaN
9           Hainan          1  2020-01-22          4           NaN
10          Shanxi          1  2020-01-22          1           NaN
11          Yunnan          1  2020-01-22          1           NaN
12         Beijing          1  2020-01-22         14           NaN
13         Guangxi          1  2020-01-22          2           NaN
14         Guizhou          1  2020-01-22          1           NaN
15         Jiangsu          1  2020-01-22          1           NaN
16         Jiangxi          1  2020-01-22          2           NaN
17         Ningxia          1  2020-01-22          1           NaN
18         Qinghai          1  2020-01-22          0           NaN
19         Shaanxi          1  2020-01-22          0           NaN
20         Sichuan          1  2020-01-22          5           NaN
21         Tianjin          1  2020-01-22          4           NaN
22        Liaoning          1  2020-01-22          2           NaN
23        Shandong          1  2020-01-22          2           NaN
24        Shanghai          1  2020-01-22          9           NaN
25        Xinjiang          1  2020-01-22          0           NaN
26        Zhejiang          1  2020-01-22         10           NaN
27       Chongqing          1  2020-01-22          6           NaN
28       Guangdong          1  2020-01-22         26           NaN
29    Heilongjiang          1  2020-01-22          0           NaN
30  Inner Mongolia          1  2020-01-22          0           NaN
31           Anhui          2  2020-01-23          9           8.0
32           Gansu          2  2020-01-23          2           2.0
33           Hebei          2  2020-01-23          1           0.0
34           Henan          2  2020-01-23          5           0.0
35           Hubei          2  2020-01-23        444           0.0
36           Hunan          2  2020-01-23          9           5.0
37           Jilin          2  2020-01-23          1           1.0
38           Tibet          2  2020-01-23          0           0.0
39          Fujian          2  2020-01-23          5           4.0
40          Hainan          2  2020-01-23          5           1.0
41          Shanxi          2  2020-01-23          1           0.0
42          Yunnan          2  2020-01-23          2           1.0
43         Beijing          2  2020-01-23         22           8.0
44         Guangxi          2  2020-01-23          5           3.0
45         Guizhou          2  2020-01-23          3           2.0
46         Jiangsu          2  2020-01-23          5           4.0
47         Jiangxi          2  2020-01-23          7           5.0
48         Ningxia          2  2020-01-23          1           0.0
49         Qinghai          2  2020-01-23          0           0.0

In [ ]:
%%bigquery
select state, date, confirmed,daily_growth
from (WITH cte AS(SELECT state, date_rank, date, confirmed,
            FROM covid_19_staging.cases_in_China )
SELECT c1.*, c1.confirmed - c2.confirmed as daily_growth
FROM cte c1
LEFT JOIN cte c2 ON c1.state = c2.state AND c2.date_rank = c1.date_rank - 1)
where state = "Hubei" order by daily_growth desc limit 10

Therefore, on Feb 13th Hubei province has the most daily increase

In [20]:
%%bigquery
select date, sum(daily_growth) as daily_growth
from (
WITH cte AS(SELECT state, date_rank, date, confirmed,
            FROM covid_19_staging.cases_in_China )
SELECT c1.*, c1.confirmed - c2.confirmed as daily_growth
FROM cte c1
LEFT JOIN cte c2 ON c1.state = c2.state AND c2.date_rank = c1.date_rank - 1
)
group by date
order by daily_growth desc

date  daily_growth
0    2020-02-13       15133.0
1    2020-02-14        6460.0
2    2020-02-02        4727.0
3    2020-02-04        3987.0
4    2020-02-05        3729.0
..          ...           ...
99   2020-01-22           NaN
100  2020-03-24           NaN
101  2020-03-26           NaN
102  2020-03-29           NaN
103  2020-03-30           NaN

[104 rows x 2 columns]

Therefore, Feb 13th also has the most increased cases nationalwide

In [89]:
%%bigquery
with cte as(select ROW_NUMBER() over(ORDER BY date) as date_rank, date, sum(confirmed) as confirmed,
            FROM covid_19_staging.cases_in_China
           group by date)
SELECT c1.*, round((c1.confirmed - c2.confirmed)*100/c2.confirmed,2)  as daily_growth_rate
FROM cte c1
LEFT JOIN cte c2 ON c2.date_rank = c1.date_rank - 1

date_rank        date  confirmed  daily_growth_rate
0           89  2020-04-24      77609            5193.93
1           64  2020-03-30      78087            1601.24
2           94  2020-04-29      83940            1446.71
3           63  2020-03-29       4590             177.51
4           62  2020-03-26       1654             146.13
..         ...         ...        ...                ...
99          54  2020-03-15      73594              -6.41
100         93  2020-04-28       5427             -92.83
101         88  2020-04-23       1466             -98.09
102         61  2020-03-24        672             -99.09
103          1  2020-01-22        547                NaN

[104 rows x 4 columns]

visualization - case count trend in China

In [16]:
%%bigquery
create view reporting.v_case_count_trend_China as
select state,last_update,confirmed,deaths,recovered  from 
myproject60392.covid_19_staging.cases_SQL_Final 
where country in ('Mainland China', 'China')

Empty DataFrame
Columns: []
Index: []

visualization - case count trend in Wuhan comparing to other provinces

In [18]:
%%bigquery
create view reporting.v_case_count_trend_Hubei_vs_other as
select "Hubei" as province, cast(last_update as date) as date,sum(confirmed) as confirmed, sum(deaths) as deaths, sum(recovered) as recovered
from myproject60392.covid_19_staging.cases_SQL_Final
where state = "Hubei"
group by date
union all
select "Provinces other than Hubei" as province, cast(last_update as date) as date,sum(confirmed) as confirmed, sum(deaths) as deaths, sum(recovered) as recovered
from myproject60392.covid_19_staging.cases_SQL_Final
where country in ('Mainland China', 'China')  and state != "Hubei"
group by date
order by date

Empty DataFrame
Columns: []
Index: []

Jan 23rd Hubei placed under quarantine - national wide case count

In [23]:
%%bigquery
create view reporting.v_case_count_China_1_23 as
select * from myproject60392.covid_19_staging.cases_in_China
where date ="2020-01-23"

Empty DataFrame
Columns: []
Index: []

April 8th Hubei travel restrictions lifted - national wide case count

In [24]:
%%bigquery
create view reporting.v_case_count_China_4_8 as
select * from myproject60392.covid_19_staging.cases_in_China
where date ="2020-04-08"

Empty DataFrame
Columns: []
Index: []

Feb 13th most increased date - national wide case count

In [25]:
%%bigquery
create view reporting.v_case_count_China_2_13 as
select * from myproject60392.covid_19_staging.cases_in_China
where date ="2020-02-13"

Empty DataFrame
Columns: []
Index: []

the trend of daily increased rate in China

In [88]:
%%bigquery
create view reporting.v_daily_growth_China as
with cte as(select ROW_NUMBER() over(ORDER BY date) as date_rank, date, sum(confirmed) as confirmed,
            FROM myproject60392.covid_19_staging.cases_in_China
           group by date)
SELECT c1.*, round((c1.confirmed - c2.confirmed)*100/c2.confirmed,2)  as daily_growth_rate
FROM cte c1
LEFT JOIN cte c2 ON c2.date_rank = c1.date_rank - 1

Empty DataFrame
Columns: []
Index: []